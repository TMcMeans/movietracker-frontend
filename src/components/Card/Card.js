import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class Card extends Component {
  constructor() {
    super();
    this.state = {
      isFavorite: false
    };
  }

  handleClick = () => {
    if (!this.state.isFavorite) {
      this.props.addFavorite(this.props.movie, this.props.user.id);
    }
    this.setState({
      isFavorite: !this.state.isFavorite
    });
  };

  render() {
    const {
      title,
      release_date,
      overview,
      movie_id,
      poster_path
    } = this.props.movie;
    return (
      <div
        className="card"
        key={movie_id}
        style={{ backgroundImage: `URL(${poster_path})` }}
      >
        <button onClick={this.handleClick}>
          <i className="far fa-heart" />
        </button>
        <div>
          <h1>{title}</h1>
          <p>{release_date}</p>
          <p>{overview}</p>
        </div>
      </div>
    );
  }
}

Card.propTypes = {
  addFavorite: PropTypes.func.isRequired,
  movie: PropTypes.object.isRequired,
  user: PropTypes.object
};
