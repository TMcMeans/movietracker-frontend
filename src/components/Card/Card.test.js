import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card';

describe('Card', () => {
  const mockMovie = {
    movie_id: 381288,
    overview:
      'Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct threâ€¦',
    poster_path:
      'https://image.tmdb.org/t/p/w500////rXMWOZiCt6eMX22jWuTOSdQ98bY.jpg',
    release_date: '2017-01-19',
    title: 'Split',
    vote_average: 7.2
  };
  const mockUser = {
    id: 1
  };
  const wrapper = shallow(
    <Card
      movie={mockMovie}
      key={mockMovie.title}
      addFavorite={jest.fn}
      user={mockUser}
    />
  );
  it('should render snapshot with correct data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addFavorite on click', () => {
    //Test nested function call to addFavorite??
  });

  it('should change state if isFavorite to true on handleClick', () => {
    const expected = {
      isFavorite: true
    };

    wrapper.instance().handleClick();
    expect(wrapper.state()).toEqual(expected);
  });
});
