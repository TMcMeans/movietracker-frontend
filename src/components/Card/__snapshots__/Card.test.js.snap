// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render snapshot with correct data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[Function]}
    movie={
      Object {
        "movie_id": 381288,
        "overview": "Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct thre…",
        "poster_path": "https://image.tmdb.org/t/p/w500////rXMWOZiCt6eMX22jWuTOSdQ98bY.jpg",
        "release_date": "2017-01-19",
        "title": "Split",
        "vote_average": 7.2,
      }
    }
    user={
      Object {
        "id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "381288",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
        >
          <i
            className="far fa-heart"
          />
        </button>,
        <div>
          <h1>
            Split
          </h1>
          <p>
            2017-01-19
          </p>
          <p>
            Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct thre…
          </p>
        </div>,
      ],
      "className": "card",
      "style": Object {
        "backgroundImage": "URL(https://image.tmdb.org/t/p/w500////rXMWOZiCt6eMX22jWuTOSdQ98bY.jpg)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="far fa-heart"
          />,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "far fa-heart",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Split
            </h1>,
            <p>
              2017-01-19
            </p>,
            <p>
              Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct thre…
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Split",
            },
            "ref": null,
            "rendered": "Split",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2017-01-19",
            },
            "ref": null,
            "rendered": "2017-01-19",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct thre…",
            },
            "ref": null,
            "rendered": "Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct thre…",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "381288",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
          >
            <i
              className="far fa-heart"
            />
          </button>,
          <div>
            <h1>
              Split
            </h1>
            <p>
              2017-01-19
            </p>
            <p>
              Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct thre…
            </p>
          </div>,
        ],
        "className": "card",
        "style": Object {
          "backgroundImage": "URL(https://image.tmdb.org/t/p/w500////rXMWOZiCt6eMX22jWuTOSdQ98bY.jpg)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="far fa-heart"
            />,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "far fa-heart",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Split
              </h1>,
              <p>
                2017-01-19
              </p>,
              <p>
                Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct thre…
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Split",
              },
              "ref": null,
              "rendered": "Split",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2017-01-19",
              },
              "ref": null,
              "rendered": "2017-01-19",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct thre…",
              },
              "ref": null,
              "rendered": "Though Kevin has evidenced 23 personalities to his trusted psychiatrist, Dr. Fletcher, there remains one still submerged who is set to materialize and dominate all the others. Compelled to abduct thre…",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
